### PR TITLE
Make notary flow return a collection of signatures to support the BFT…

### PR DESCRIPTION
**Pull request by adagys**
… notary. For a single-node or RAFT notary it would just contain a single signature.

### CODE REVIEW

File: core/src/main/kotlin/net/corda/flows/NotaryFlow.kt [hunk 1]
```diff
@@ -23,7 +23,7 @@ object NotaryFlow {
      *                         by another transaction or the timestamp is invalid.
      */
     open class Client(private val stx: SignedTransaction,
-                      override val progressTracker: ProgressTracker) : FlowLogic<DigitalSignature.WithKey>() {
+                      override val progressTracker: ProgressTracker) : FlowLogic<Collection<DigitalSignature.WithKey>>() {
```
Comment (shamsasari) [inline core/src/main/kotlin/net/corda/flows/NotaryFlow.kt]: `List` is a better abstraction than `Collection`

File: samples/trader-demo/src/main/kotlin/net/corda/traderdemo/flow/SellerFlow.kt [hunk 2]
```diff
@@ -81,7 +81,7 @@ class SellerFlow(val otherParty: Party,
 
             // Get the notary to sign the timestamp
             val notarySig = subFlow(NotaryFlow.Client(tx.toSignedTransaction(false)))
```
Comment (shamsasari) [inline samples/trader-demo/src/main/kotlin/net/corda/traderdemo/flow/SellerFlow.kt]: `notarySigs`

File: core/src/main/kotlin/net/corda/flows/NotaryFlow.kt (hunk 1 above)
Comment (adagys) [inline core/src/main/kotlin/net/corda/flows/NotaryFlow.kt]: how come?

File: samples/trader-demo/src/main/kotlin/net/corda/traderdemo/flow/SellerFlow.kt (hunk 2 above)
Comment (adagys) [inline samples/trader-demo/src/main/kotlin/net/corda/traderdemo/flow/SellerFlow.kt]: I was wondering if we can treat a list of signatures as conceptually a single "notary signature", but that's probably more confusing...

File: core/src/main/kotlin/net/corda/flows/NotaryFlow.kt (hunk 1 above)
Comment (shamsasari) [inline core/src/main/kotlin/net/corda/flows/NotaryFlow.kt]: It's ordered while `Collection.iterator` gives no such guarantee. Also the `get(index)` will be almost necessary for those who know they're not using a BFT notary and can thus simply do `.get(0)`.

File: core/src/main/kotlin/net/corda/flows/NotaryFlow.kt [hunk 3]
```diff
@@ -23,7 +23,7 @@ object NotaryFlow {
      *                         by another transaction or the timestamp is invalid.
```
Comment (shamsasari) [inline core/src/main/kotlin/net/corda/flows/NotaryFlow.kt]: This doc needs to be updated as it's still in the singular.

Comment (shamsasari) [inline core/src/main/kotlin/net/corda/flows/NotaryFlow.kt]: And with some brief explanation why the notary can send multiple sigs.

**Comment by adagys** [comment]
Ready for another look

**Pull request merged**